tableaudocumentapi/datasource.py

```diff
@@ -59,6 +59,29 @@ def from_file(cls, filename):
         dsxml = ET.parse(filename).getroot()
         return cls(dsxml, filename)
 
+    def get_connections_by_server_name(self, server_name):
+        matches = []
+        if server_name:
+            for connection in self._connections:
+                if str(server_name) == connection.server:
+                    matches.append(connection)
+
+        return matches
+
+    def get_connections_by_attributes(self, attribs):
```

Comment:
This is a different approach, instead of a method for a specific attribute you can pass in a dict of attributes and it'll match on any of them.